lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -90,6 +91,38 @@ def record(formatted_message, severity)
         nil
       end
 
+      def record_logstasher_event(log)
+        return unless logstasher_enabled?
+
+        severity = log['level'] || 'UNKNOWN'
+        return if log.key?('message') && (log['message'].nil? || log['message'].empty?)
+
+        if NewRelic::Agent.config[METRICS_ENABLED_KEY]
```

Comment:
It seems like much of the logic here: incrementing the seen and severity counts, checking for forwarding enabled, checking for high security, getting a transaction and its priority, etc. is shared between the `Logger` and `LogStasher` instrumentations. Perhaps some of it could go into helper methods that are shared by both.

Reply:
I like this! Created some shared methods [here](https://github.com/newrelic/newrelic-ruby-agent/pull/2559/commits/4fabc5bb9191ae478d560eebb9fce9e0d2b03dd3) and [here](https://github.com/newrelic/newrelic-ruby-agent/pull/2559/commits/229c7d1a80992e3d80c52d087cb9add67f5dda3b)